### ISSUE TITLE
Modularize permission classes

### ISSUE THREAD

**Issue by Neraste** [DakaraProject/dakara-server#74]
### Modularize permission classes

Permission classes are too specific and heavy to use. This is due to the fact that when a list of them is used in a view, they are implicitly chained by an And operator. Moreover, some permission classes affect only some methods, which makes their reuse complicated or their name clumsy. It would be interesting to use a third-party library that allows to use other operators to obtain complex permission classes per view. This would result in smaller, atomic permission classes that are easier to manipulate.

The documentation of DRF about permission lists several third-party libraries for such a task. Among them, [REST condition](https://github.com/caxap/rest_condition) allows to simply use And, Or and Not operator, which is enough for our needs.

By instance, in `PlaylistEntryListView`:
```python
permission_classes = [
    IsPlaylistUserOrReadOnly,
    IsPlaylistAndLibraryManagerOrSongCanBeAdded,
    KaraokeIsNotStoppedOrReadOnly,
]                                                           
```
Becomes:
```python
permission_classes = [Or(
    RequestIsReadOnly,
    And(
        KaraokeIsNotStopped,
        Or(
            And(IsPlaylistUser, SongCanBeAdded),
            And(IsPlaylistManager, IsLibraryManager, Not(SongCanBeAdded))
        )
    )
)]
```
which is longer, but easier to read.

This [example](https://stackoverflow.com/a/21428671/4584444) shows how to combine atomic permission classes to achieve per-method permissions. I think that the basic permission classes should be put in common in the Internal app.

### Remove some permission classes

Moreover, after a close reading of the DRF permission classes documentation, it appears that the role of this mechanism is to tell, before anything is done (*i.e.* before deserialization), whether *this* user has the right to call *this* view method on *this* object, if any. Any other condition should be handled by another mechanism, in the view itself. The playlist maximum limit condition is done accordingly, whereas the karaoke being stopped condition is performed through DRF permission classes, which is not valid.

**Comment by Neraste** [comment]
Factorize the methods to check user permission outside of permission classes. Those methods should be tested as well.

**Comment by Neraste** [comment]
This modularization should be applied on the front-end as well.

**Comment by Neraste** [comment]
Django REST Framework v3.9.0 allows to compose permission classes, but there is a bug that prevent to compose compositions (i.e. `(A & B) & (C & D)` fails). The v3.9.1 will solve this problem.

**Comment by Nadeflore** [comment]
Fixed by #93.
